webpack/assets/javascripts/react_app/components/Layout/Layout.js

```diff
@@ -85,9 +85,7 @@ class Layout extends React.Component {
           onItemClick={primary =>
             handleMenuClick(primary, activeMenu, changeActiveMenu)
           }
-          onNavigate={({ href, preventHref }) =>
```

Comment:
The `preventHref` was here so we can use turbolinks when needed, looks like it can be removed from more places now.

Reply:
Could you point me to an usage of this method? I can't see where we use it, or you mean the `preventHref` is used on different places as well? If so, could you point me to those? :) I found it only in `LayoutSelectors.js` something I've missed?

Reply:
1. The `LayoutSelectors.js` uses `item.turbolinks` received from the server:
https://github.com/theforeman/foreman/blob/develop/webpack/assets/javascripts/react_app/components/Layout/LayoutSelectors.js#L33
Not sure if this PR should remove it, if so, there are also the `data-turbolinks-track` we use in many places.
There is a menu-items registry where we define menu items to use turbolinks :-1: 
https://github.com/theforeman/foreman/blob/develop/app/registries/menu/item.rb#L21
Looks like `turbolinks` is default true, we just use `turbolinks` for everything.

2. Looks like the `preventHref` still needed so it will use the `onNavigate` for the navigation.
See: https://github.com/patternfly/patternfly-react/blob/dbff0f181a2bd6fd763fb6769e8f9ef49281178d/packages/patternfly-3/patternfly-react/src/components/VerticalNav/VerticalNavItemHelper.js

Reply:
@ezr-ondrej looks like this comment is still not addressed? or am i missing something?

Reply:
oh, I've missed this one

Reply:
@sharvit we don't need to use the `preventHref` right?

Reply:
Shouldn't we set it to true?
If not the browser will handle it natively? see this code from inside pf:
```js
if (preventHref && !!onClick) {
  event.preventDefault();
}
```
https://github.com/patternfly/patternfly-react/blob/dbff0f181a2bd6fd763fb6769e8f9ef49281178d/packages/patternfly-3/patternfly-react/src/components/VerticalNav/VerticalNavItemHelper.js#L88-L90

Reply:
I thought we want that for pages not handled by react router, right? :thinking:

Reply:
**I'm confused already but I think we should set it to `true`.**

We need the `preventDefault` when clicking a link so instead of using the native browser `href` mechanism, it would use the react-router.

If you would set it to `false`, wouldn't it use react-router while also using the native browser link?

Reply:
so what you are suggesting is we don't actually need to pass it in to the `onNavigate` function (since it isn't using it anyways) but rather explicitly add another prop below that does
```jsx
preventDefault=true
```
instead of relying on `{...this.props}` to include it and pass it on?

Reply:
Yes!